document/src/vespa/document/repo/documenttyperepo.cpp

```diff
@@ -502,6 +506,329 @@ void configureAllRepos(const DocumenttypesConfig::DocumenttypeVector &t, Documen
     }
 }
 
+using DataTypesByIdx = hash_map<int, const DataType *>;
+using StructTypesByIdx = hash_map<int, StructDataType *>;
+
+const StructDataType * performStructInherit(int idx,
+                                            const DocumenttypesConfig::DoctypeVector &t,
+                                            const StructTypesByIdx &structs)
+{
+    for (const auto & docT : t) {
+        for (const auto & structT : docT.structtype) {
+            if (idx == structT.idx) {
+                StructDataType *st = structs[idx];
+                for (const auto & inheritD : structT.inherits) {
+                    const auto * parent = performStructInherit(inheritD.type, t, structs);
+                    if (parent == nullptr) {
+                        LOG(error, "Missing parent type [idx %d] for struct %s",
+                            inheritD.type, structT.name.c_str());
+                        throw IllegalArgumentException("missing parent type");
+                    }
+                    LOG_ASSERT(st != nullptr);
+                    for (const auto & field : parent->getFieldSet()) {
+                        st->addInheritedField(*field);
+                    }
+                }
+                return st;
+            }
+        }
+    }
+    return nullptr;
+}
+
+void configureDocTypes(const DocumenttypesConfig::DoctypeVector &t, DocumentTypeMap &type_map) {
```

Comment:
This a rather large function. I think it would be more readable if it was split into smaller parts.
Perhaps 1 per for loop or